suite-native/atoms/src/Button/Button.tsx

```diff
@@ -227,9 +238,32 @@ export const ButtonIcon = ({ iconName, color, buttonSize }: ButtonIconProps) =>
     <Icon name={iconName} color={color} size={buttonToIconSizeMap[buttonSize]} />
 );
 
+export const isIconName = (element: ButtonAccessoryType) =>
+    G.isNotNullable(element) &&
+    typeof element === 'string' &&
+    icons[element as IconName] !== undefined;
+
+// ButtonAccessoryView renders either a ButtonIcon or a provided custom element
+// iconColor and iconSize are only used when element is an IconName
+export const ButtonAccessoryView = ({
+    element,
+    iconColor = 'iconDefault',
+    iconSize = 'medium',
+}: ButtonAccessoryViewType) => {
+    if (isIconName(element)) {
+        if (!iconColor || !iconSize) return null;
+
+        return (
+            <ButtonIcon iconName={element as IconName} color={iconColor} buttonSize={iconSize} />
```

Comment:
Change the `isIconName` to a type guard (`const isIconName = (value: ButtonAccessoryType) : is IconName => ...`) more here: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards. 

Then you can remove the type assertion here.

```suggestion
            <ButtonIcon iconName={element} color={iconColor} buttonSize={iconSize} />
```

Reply:
0c52c10957